tests/src/Context/ContextUtil.php

```diff
@@ -11,6 +11,22 @@
  */
 trait ContextUtil {
 
+  /**
+   * Determine if the a user is already logged in.
+   *
+   * Override the existing loggedIn method from DrupalExtension,
+   * to skip login form test in user/login page
+   * (because ecas revokes permissions for this page).
+   */
+  public function loggedIn() {
```

Comment:
It is an idea based of the current code review and the comments put in the PR and in the code. It is not a comment that could block the PR if the idea is feasible but I would like your opinion on it.

So, we override the existing loggedIn method of DrupalExtension which is the `Drupal\DrupalExtension\Context\DrupalContext` which extends `Drupal\DrupalExtension\Context\RawDrupalContext`.
This class has the method `getUserManager()` to get a `\Drupal\DrupalUserManagerInterface` implementation object.

This object has a method `getCurrentUser()` that returns FALSE if the user is anonymous; I.E. not logged.

To avoid to be dependent of a a visual element that could change if a contrib module or a Drupal core code could change it, it is would not be better to use the `getCurrentUser()` method to test if the user is logged or not? If it returns FALSE, that means the user is not logged.

Of course, if the final goal is to use it outside the Drupal related context objects, forget this idea.

Reply:
I had the same thought, Gilles. I also don't understand why it was done this way. 
I've based my decision on:
* Keep the logic aligned with the widely used "drupal-extension";
* Keep the logic as it was before, our goal is just updating a package here and not exactly improving the tests.

I'm curious to understand why this was done like this. But this discussion should occur first in the "drupal-extension" scope.